assignment/core/src/main/java/com/assignment/core/schedulers/WorkflowScheduler.java

```diff
@@ -0,0 +1,156 @@
+
+
+    @Reference
+    private Scheduler scheduler;
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    private String csvPath;
+    private int schedulerId;
+    private String model;
+
+    /**
+     * Activates the scheduler with the provided configuration.
+     *
+     * @param configuration The configuration to apply.
+     */
+    @Activate
+    private void activate(WorkflowSchedulerConfiguration configuration) {
+        logger.info("Activating Workflow Scheduler with configuration: {}", configuration);
+        this.csvPath = configuration.csvPath();
+        this.schedulerId = configuration.schedulerName().hashCode();
+        this.model = configuration.model();
+        addScheduler(configuration);
+    }
+
+    /**
+     * Updates the scheduler with new configuration.
+     *
+     * @param configuration The updated configuration.
+     */
+    @Modified
+    protected void modified(WorkflowSchedulerConfiguration configuration) {
+        logger.info("Modifying Workflow Scheduler with new configuration: {}", configuration);
+        removeScheduler();
+        this.csvPath = configuration.csvPath();
+        this.schedulerId = configuration.schedulerName().hashCode();
+        this.model = configuration.model();
+        addScheduler(configuration);
+    }
+
+    /**
+     * Deactivates and removes the scheduler.
+     *
+     */
+    @Deactivate
+    protected void deactivate() {
+        logger.info("Deactivating Workflow Scheduler.");
+        removeScheduler();
+    }
+
+    /**
+     * Adds a scheduler with the specified configuration.
+     *
+     * @param config The configuration to use for scheduling.
+     */
+    private void addScheduler(WorkflowSchedulerConfiguration config) {
+        boolean enabled = config.enabled();
+        if (enabled) {
+            ScheduleOptions scheduleOptions = scheduler.EXPR(config.cronExpression());
+            scheduleOptions.name(String.valueOf(schedulerId));
+            scheduleOptions.canRunConcurrently(false);
+            scheduler.schedule(this, scheduleOptions);
+            logger.info("Workflow Scheduler added successfully with name: {}", schedulerId);
+
+            ScheduleOptions scheduleOptionsNow = scheduler.NOW();
+            scheduler.schedule(this, scheduleOptionsNow);
+        } else {
+            logger.info("Workflow Scheduler is in disabled state.");
+        }
+    }
+
+    /**
+     * Removes the scheduler.
+     *
+     */
+    private void removeScheduler() {
+        logger.info("Removing Workflow Scheduler with name: {}", schedulerId);
+        scheduler.unschedule(String.valueOf(schedulerId));
+    }
+
+    /**
+     * Executes the workflow based on the configuration.
+     */
+    @Override
+    public void run() {
+        ResourceResolver resourceResolver = null;
+        try {
+            Map<String, Object> param = new HashMap<>();
+            param.put(ResourceResolverFactory.SUBSERVICE, "workflowservice");
```

Comment:
how will be subservice and user mapping happen at AEM if a fresh AEM Instance is used? Explore ensure service user in aem to create service user and use them to create service users from code.